frontend/src/containers/NavigationTree.js

```diff
@@ -107,13 +107,15 @@ class NavigationTree extends Component {
   };
 
   queryRequest = async (value) => {
+    this.setState({ pendingQuery: true });
     try {
       const response = await queryPathsRequest(value, '', true);
 
       await new Promise((resolve) =>
         this.setState(
```

Comment:
pls implement the same protection mechanism to make sure we get the last result.
(like [here](https://github.com/metasfresh/metasfresh/pull/12380/files#r789665656))

Reply:
added

Reply:
pls apply the same pattern u did on MenuOverlay.js .